development/webpack/utils/plugins/ManifestPlugin/helpers.ts

```diff
@@ -48,9 +104,14 @@ export function transformManifest(args: { lockdown: boolean; test: boolean }) {
 
   return transforms.length
     ? (browserManifest: chrome.runtime.Manifest, _browser: string) => {
-        const clone = structuredClone(browserManifest);
```

Comment:
The function is working correctly (the Result shows the correct merged object), but the problem is that it's possible  `browserManifest ` can be an empty object. `transform(clone)` modifies the clone in place, but `addManifestFlags` returns a new object instead of modifying the input. We need to capture the return value. This change ensures that if a transform returns a new object (like `addManifestFlags` does), and that returned object is used for subsequent transforms.

Reply:
ah, interesting. I think a better way is to mutate `browserManifest` instead. So instead of  `return merge({}, browserManifest, manifestFlags);` use `merge(browserManifest, manifestFlags);` (do NOT `return` anything from this function). This way you won't need this extra logic or type changes and the code will technically be faster since it doesn't create and new copy properties to a new object (though doubt it's measurable).

These functions are supposed to mutate `browserManifest`.

Reply:
Thanks for suggestions, this works for me ! modified in a199f6fa9e7d2af314a91f46aee7f01981d48f20